### ISSUE TITLE
[Python] Implement new Concept Exercise: rich comparisons

### ISSUE THREAD

**Issue by DavidGerva** [exercism/python#2287]
This issue describes how to implement the `rich comparisons` concept exercise for the python track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

This concept exercise is meant to teach an understanding/use of `rich comparison` methods in python. It should also generally explain how to override the behavior of `rich comparison` _operators_.

## Learning objectives

- understanding (_and using the concept_) that the `==` operator calls the dunder method `__eq__()` on a specific object, and uses that object's implementation for comparison.  Where no implementation is present, the default `__eq__()` from _generic_ `object` is used.
- the same applies for operators `<`, `<=`, `!=`, `>=`, `>` and correspondents dunder methods `__lt__`, `__le__`, `__ne__`, `__ge__`, `__gt__`
- overriding the default implementation of the `__eq__()` dunder method on a specific object to customize comparison behavior.
- overriding the default implementations of the `__lt__`, `__le__`, `__ne__`, `__ge__`, `__gt__` dunder methods on a specific object to customize comparison behavior.
- return the singleton `NotImplemented` if the called operation is not implemented

## Out of scope

- performance considerations

## Concepts

- `__eq__`, `__lt__`, `__le__`, `__ne__`, `__ge__`, `__gt__` dunder methods
- comparison behavior customization
- dunder methods override

## Prerequisites

- Method overriding
- Comparison priority in Python
- Comparison operators`==`, `>`, `<`, `!=`
- Identity methods `is` and `is not`
- Equality applied to `built-in` types
- Equivalence vs equality
- Inequality
- Use the _value comparison operators_ `==`, `>`, `<`, `!=` with _numeric types_
- Use the _value comparison operators_ `==`, `>`, `<`, `!=` with _non-numeric types_
- `basics`
- `booleans`
- `classes`
- `numbers`
- `sequences`
- `iteration`

## Resources to refer to

- [Rich comparison (Python language reference)](https://docs.python.org/3/reference/datamodel.html#object.__lt__)
- [Rich comparison on handshake](http://eng.handshake.com/python/2014/07/02/rich-comparison-methods/)
- [Rich comparisons PEP207](https://www.python.org/dev/peps/pep-0207/)

### Hints

- Referring to one or more of the resources linked above, or analogous resources from a trusted source.

## Concept Description 
(_a variant of this can be used for the `v3/languages/python/concepts/<concept>/about.md` doc and this exercises  `introduction.md` doc._)

_**Concept Description Needs to Be Filled In Here**_

## Representer

No changes required.

## Analyzer

TBD

## Implementing

Tests should be written using unittest.TestCase and the test file named rich_comparisons_test.py.

Code in the `.meta/example.py` file should only use syntax & concepts introduced in this exercise or one of its prerequisites. 
Please do not use comprehensions, generator expressions, or other syntax not previously covered.  Please also follow [PEP8](https://www.python.org/dev/peps/pep-0008/) guidelines.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue, or contact one of the maintainers on our Slack channel.

**Comment by BethanyG** [comment]
@DWalz  -- Please note that the description of this issue has changed, and is now linked to new requirements and documentation.  Since it's been a bit since we've heard from you, I'm going to flag this issue `[help wanted]`.  If you are still working/want to work on this issue, please let me know, and I will re-flag it as `[claimed]` for you.

**Comment by BethanyG** [comment]
This should stay open.

**Comment by BethanyG** [comment]
Flagging as close and re-write with new issue template.  Also take a look at scope.
